k8s/upgrade-job/src/helm/upgrade.rs

```diff
@@ -112,8 +112,25 @@ impl HelmUpgradeBuilder {
         // <chart-name>-<chart-version> string.
         let umbrella_chart_regex = format!(r"^({UMBRELLA_CHART_NAME}-[0-9]+\.[0-9]+\.[0-9]+)$");
         // Accepts pre-release and release, both.
+        // Q: How do I read this regex?
+        // A: This regular expressions is bounded by the '^' and '$' characters, which means
+        //    that the input string has to match all of the expression exactly. It is not enough
+        //    if a substring within the input string matches the regular expression. The pattern
+        //    requires the following conditions to be met:
+        //    1. The string must start with the literal contained in the literal CORE_CHART_NAME.
+        //       e.g.: mayastor-2.2.0 starts with 'mayastor'
+        //    2. A '-' followed by three sets of numbers (each with one or more) separated by '.',
+        //       must sit after the value of CORE_CHART_NAME.
+        //       e.g. mayastor-4.56.789 is a valid chart-name.
+        //    3. A '-' followed by one or many alphanumeric characters may optionally sit after a
+        //       chart-name like 'mayastor-1.2.3'.
+        //       e.g.: mayastor-1.2.3-testing, mayastor-1.2.3-testing-upgrade-23-35-25-05-2023,
+        //       mayastor-2.3.0-rc-3
+        //    4. The optional group of character(s) mentioned in (3) above, may optionally contain
+        //       a '.' followed by a set of numbers.
+        //       e.g.: mayastor-2.3.4-rc.1, mayastor-2.3.4-alpha.2
         let core_chart_regex =
-            format!(r"^({CORE_CHART_NAME}-[0-9]+\.[0-9]+\.[0-9]+(-rc\.[0-9]+)?)$");
+            format!(r"^({CORE_CHART_NAME}-[0-9]+\.[0-9]+\.[0-9]+(-[a-zA-Z0-9]+(\.[0-9]+)?)*)$");
```

Comment:
We can avoid all these comments and complicated regex by having 2 steps:
1. match {CORE_CHART_NAME}-{semver}
2. use semver crate to validate semver

Reply:
Hmm. This will indeed make it more readable. Just going with the regex for now, I'll refactor this part if we run into versioning issues later down the line.